src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -261,6 +297,10 @@ public void visitToken(DetailAST ast) {
         if (isInterfaceOrAnnotationMember(ast)) {
             processInterfaceOrAnnotation(ast);
         }
+
+        if (javaLanguageLevel.getIntValue() >= JavaLanguageLevel.LEVEL_17.getIntValue()) {
```

Comment:
We can make a method within the enum to make this read nicely and abstract away the fact that these are backed by integers (or maybe even better: combine these two conditions into a GTE sort of thing)?
```suggestion
        if (javaLanguageLevel == JavaLanguageLevel.LEVEL_17 || javaLanguageLevel.isGreaterThan(JavaLanguageLevel.LEVEL_17)) {
```

Reply:
Yes I think it will be better if we have a GTE method this will always be our check. 
is level >= 17 ? is level >= 21 ?  is level >= 30 ?

something like: 
  ```
      private boolean isGreaterThanOrEqual(JavaLanguageLevel level) {
            return this.getIntValue() >= level.getIntValue();
        }

```

Reply:
yep, sounds good

Reply:
fwiw, I tend to do `private boolean isGreaterThanOrEqual(JavaLanguageLevel that) ` on instance methods to make things super obvious (another good name is `other`).